### DISCUSSION TITLE
Readme.md version displayed not aligning with version selected

### DISCUSSION THREAD

**Discussion by whtsthedeal** [verdaccio/verdaccio#3899] [Q&A] [answered]
Within our registry we have versioned packages that each have readme.md files embedded containing details for that version. Upon selection of a package version within the verdaccio registry dashboard we expect that versions readme.md to then be presented, however that is not the case. Regardless of which version is selected within the verdaccio registry dashboard, the readme.md shown is always the one from the most recent package published. Is this something that is expected,  as it doesn't seem natural for other items on the registry page to reflect details for a selected version except for the readme.md? Thank you for the help!

**Comment by juanpicado** [comment, marked as answer]
This is by design, readme content takes much space, at this point all versions readme are discarded and only the latest is stored. I've been thinking for a while who to make it right without affect the size of the storage but no much research so far.

**Reply by whtsthedeal** [reply]
Ok, thank you for the insight, that is definitely a bummer, but I understand your reasoning for avoiding the overhead of storing that for each package version. 

Moving forward if there was any progress on this feature, potentially an option within the configuration file to toggle this feature for users who can manage the storage overhead versus those who can't? That way it's up to the user to handle that problem given their readme size and desire to have the latest readme shown versus the selected version's readme.  

Just spitballing some solutions as I could see others wanting this feature as well, thank you again for the information.

**Comment by juanpicado** [comment]
> Ok, thank you for the insight, that is definitely a bummer, but I understand your reasoning for avoiding the overhead of storing that for each package version.
> 
> Moving forward if there was any progress on this feature, potentially an option within the configuration file to toggle this feature for users who can manage the storage overhead versus those who can't? 

Yeah, sounds reasonable, I'll think what internal impact might have, I don't think is a big change at this point.

That way it's up to the user to handle that problem given their readme size and desire to have the latest readme shown versus the selected version's readme.
> 
> Just spitballing some solutions as I could see others wanting this feature as well, thank you again for the information.

Thanks for the insights, if you have further ideas just feel free to share them.

**Comment by mbtools** [comment]
After spending some time on the readme issue (https://github.com/verdaccio/verdaccio/pull/4493), I think we should definitely change the behaviour. Projects often have several parallel releases like verdaccio. Having only one version of the readme will most likely show something incorrect if `latest` is related to a different release.

Verdaccio:

![image](https://github.com/verdaccio/verdaccio/assets/59966492/4f58a0a2-0693-4835-bd18-5e5dfbf3f683)

Semver package:

![image](https://github.com/verdaccio/verdaccio/assets/59966492/d02119f5-c272-4a04-9c33-e6ec0a395851)

I suggest storing the readme for all versions that have a `dist-tag` as a default. This would keep the number of readmes very small but provide much better user experience. 

For other versions, the web interface could show links available readmes (i.e. dist-tags). 

If a package has no readme at all, I suggest showing a user-friendly message like npm instead of "ERROR: No README data found!":

![image](https://github.com/verdaccio/verdaccio/assets/59966492/1956bece-f93b-4927-83df-ffeae6c55eb7)
